packages/libllvm/tools-clang-lib-Driver-ToolChains-Linux.cpp.patch

```diff
@@ -1,74 +1,28 @@
 --- tools/clang/lib/Driver/ToolChains/Linux.cpp.orig	2019-12-21 22:30:03.676720096 +0000
 +++ ./tools/clang/lib/Driver/ToolChains/Linux.cpp	2019-12-21 22:34:42.941719383 +0000
-@@ -316,6 +316,41 @@
+@@ -403,7 +403,7 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
+   if (Triple.getVendor() == llvm::Triple::OpenEmbedded &&
+       Triple.isArch64Bit())
+     addPathIfExists(D, SysRoot + "/usr/" + OSLibDir, Paths);
+-  else
++  else if (!IsAndroid)
+     addPathIfExists(D, SysRoot + "/usr/lib/../" + OSLibDir, Paths);
+   if (IsRISCV) {
+     StringRef ABIName = tools::riscv::getRISCVABI(Args, Triple);
+@@ -451,7 +451,15 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
+     addPathIfExists(D, D.Dir + "/../lib", Paths);
  
-   const std::string OSLibDir = getOSLibDir(Triple, Args);
-   const std::string MultiarchTriple = getMultiarchTriple(D, Triple, SysRoot);
-+  bool NativeBuild = true;
+   addPathIfExists(D, SysRoot + "/lib", Paths);
+-  addPathIfExists(D, SysRoot + "/usr/lib", Paths);
++  bool nativeBuild = Triple == llvm::Triple(llvm::sys::getDefaultTargetTriple());
++  if (nativeBuild || !IsAndroid)
++    addPathIfExists(D, SysRoot + "/usr/lib", Paths);
 +
-+  if(IsAndroid) {
-+    if (MultiarchTriple == llvm::sys::getDefaultTargetTriple())
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/lib", Paths);
-+    else
-+      NativeBuild = false;
-+
-+    if (Arch == llvm::Triple::aarch64) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/aarch64-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib64", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/aarch64-linux-android/lib");
-+    }
-+    else if (Arch == llvm::Triple::arm || Arch == llvm::Triple::thumb) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/arm-linux-androideabi/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/arm-linux-androideabi/lib");
-+    }
-+    else if (Arch == llvm::Triple::x86_64) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/x86_64-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib64", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/x86_64-linux-android/lib");
-+    }
-+    else if (Arch == llvm::Triple::x86) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/i686-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/i686-linux-android/lib");
-+    }
-+
-+    ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/lib");
++  if (IsAndroid) {
++    addPathIfExists(D, SysRoot + "/usr/" + MultiarchTriple + "/lib", Paths);
++    addPathIfExists(D, "/system/" + OSLibDir, Paths);
++    ExtraOpts.push_back("-rpath=" + SysRoot + "/usr/lib");
```

Comment:
This block does the same as the larger block it replaces, with one exception: it removes the rpath for the ndk-multilib arch-specific directory. I found that the problems reported in #5031 go away if this arch-specific rpath is removed, rather than removing those shared libraries as I first thought.